### DISCUSSION TITLE
Yocto recipe question for PolarFire SoC

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#363] [Linux build systems for PolarFire SoC] [answered]
a Yocto Linux query follows.. for a customer need to build an image that combines a RAM based filesystem. (mpfs-initramfs-image) with a mtdutils image (need this format since they are loading the image over JTAG). This is an unusual setup.

I can do both steps individually so can successfully build and deploy

mpfs-initramfs-image
core-image-minimal-mtdutils.
 

Now wish to combine the 2. Of course we have individual recipe files in recipes-core but I think I have to create a special recipe file that combines these 2 or can I just use an include directive in one recipe file to include the other.

Thx
Dave

**Comment by vfalanis** [comment]
Hi Dave,

Could you please explain further why you mean by mtdutils image format? Do you mean you want a raw binary image that contains an initram filesystem? Do you need an image that runs Linux from eMMC/SD?

Thanks

**Reply by rockoguido** [reply]
hi.  the intent of have this mtdutils image format is because the only way the customer is able to get the image down to Nand flash on the target is via JTAG.      So apparently only this image format is compatible with the host tool being used to load the image down to the target.   Target has no SD card or eMMC, just NAND flash and 1 Gig of DDR
I did a test myself and was able to build this image type via bitbake core-image-minimal-mtdutils.

**Reply by vfalanis** [reply]
Hi Dave,

If I understand the requirement correctly, they would like to load the Linux boot artifacts from QSPI NAND and then boot from a RAM disk filesystem (initramfs). As you said, this is an unusual case, since most customers are looking for a flash based file system like UBI that runs directly from the NAND. 

What you described is doable but would require additional patches to U-Boot and few other parts of the Yocto BSP. In fact we added support for UBI instead of initramfs since this is the recommended and most common use case.

If the customer wants to proceed with the initramfs approach, my recommendation would be to look at the 2022.03 Yocto release and port a few changes over to the latest release (2023.09). The relevant changes can be found [here](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/commit/dec809e382415a4504277c0688a6c4c33a520820)

The commit above uses a FITimage raw partition instead of UBI, so it should be similar or the same to what the customer is asking for.

**Reply by rockoguido** [reply, marked as answer]
hi and thank you for the update.     Turns out this customer uses UBIFS on their NAND flash.   So, they've setup HSS to boot via QSPI/NAND flash.    So, they have moved away from the booting a ram disk filesystem.
We have challenges bringing up Linux on their platform but I'll start a separate thread on that.   
Thx!
Dave
